src/main/java/com/zerobase/mentalgrowhdiary/service/DiaryService.java

```diff
@@ -108,4 +111,25 @@ public List<DiaryResponseDto> searchDiary(String client, LocalDate diaryDate, St
         return diaries.stream().map(DiaryResponseDto::fromEntity).toList();
     }
 
+    @Transactional(readOnly = true)
+    public List<String> getHashtagSuggest(String client,String prefix) throws JsonProcessingException {
+
+        User user = userRepository.findByUsername(client)
+                .orElseThrow(() -> new MentalGrowthException(ErrorCode.USER_NOT_FOUND));
+
+        List<Diary> diaries = diaryRepository.findTop7ByUserOrderByDiaryDateDesc(user);
+
+        Set<String> hashtagSet = new HashSet<>();
+
+        for(Diary diary : diaries){
+            List<String> diaryHashtags = objectMapper.readValue(diary.getHashtag(), List.class);
+            hashtagSet.addAll(diaryHashtags);
+        }
+
+        return hashtagSet.stream()
+                .filter(tag -> tag.startsWith(prefix))
+                .sorted()
+                .collect(Collectors.toList());
```

Comment:
아래처럼 작성해도 동일할 것 같아요. 가능하면 변수를 final화(=재할당하지 않고, collection의 경우 add/put 등이 일어나지 않고)하여 사용하는 것이 좋습니다 (선언과 할당을 동시에)
```suggestion
        return diaries.stream().flatMap(diary -> objectMapper.readValue(diary.getHashtag(), List.class))
                                              .filter(tag -> tag.startsWith(prefix))
                                              .sorted()
                                              .collect(Collectors.toList());
```

Reply:
감사합니다. 참고하겠습니다.